drivers/bluetooth/hci/Kconfig

```diff
@@ -73,6 +73,7 @@ config BT_STM32WBA
 	depends on DT_HAS_ST_HCI_STM32WBA_ENABLED
 	select HAS_STM32LIB
 	select BT_HCI_SET_PUBLIC_ADDR
+	select TAINT_BLOBS
```

Comment:
@erwango I don't think this is how it's supposed to work. `TAINT_BLOBS' should only be selected if `west blobs fetch` has been run, and it's selected by a build-time generated Kconfig symbol through `scripts/zephyr_module.py`. I.e. there shouldn't be any hard-coded selection of it anywhere like this. @nashif or @tejlmand can correct me if I've misunderstood this.

Reply:
> @erwango I don't think this is how it's supposed to work

I'm not surprised actually. I should have add "Naive fix attempt" to the description :)

>  @nashif or @tejlmand can correct me if I've misunderstood this.

Guy, please also comment on the following points below:

> `TAINT_BLOBS` should only be selected if `west blobs fetch` has been run

If my target doesn't contain BLE, f/w won't be tainted. So, shouldn't it rather be "`TAINT_BLOBS` should only be selected if f/w is linked with blob downloaded using `west blobs fetch` " ?

> it's selected by a build-time generated Kconfig symbol through `scripts/zephyr_module.py`

It isn't the case for this target today, likely because STM32 doesn't respect the script assumptions. But what are those ? Is there a documentation that I'm missing ? The only piece of doc around this I've found is [here](https://docs.zephyrproject.org/latest/contribute/bin_blobs.html#tainting):
```
Inclusion of binary blobs will taint the Zephyr build. [...] a tainted image will be one that includes binary blobs in it.

Tainting will be communicated to the user in the following manners:

One or more Kconfig options TAINT_BLOBS_* will be set to y
```

Note also that [`TAINT_BLOBS` is selected for tflite](https://github.com/zephyrproject-rtos/zephyr/blob/fd190313aef7a8a9068837f4979f29051f4792eb/samples/modules/tflite-micro/tflm_ethosu/Kconfig#L7-L8) which would need to be fixed as well.

Reply:
> It isn't the case for this target today, likely because STM32 doesn't respect the script assumptions.

I believe it's a bug in the script that was introduced by commit 2e5c37142c3b5fb8782a7a736adc09771f14eae0. The "else" branch where there is no module Kconfig file neither in the module nor in the main tree didn't take care of the taint symbol:
https://github.com/zephyrproject-rtos/zephyr/blob/8a4b3a4b618a6b18390cfb4d67ad505abc1c83ab/scripts/zephyr_module.py#L366

This is fixed by my PR.

Reply:
> One or more Kconfig options TAINT_BLOBS_* will be set to y

That's another thing that bothers me since there is absolutely nowhere any TAINT_BLOBS_* references. The only thing there is a `TAINT_BLOBS` symbol. I.e. the documentation makes reference to non-existent symbols and doesn't make any reference to the one symbol that does exist.